codecs/plugins.go

```diff
@@ -3,12 +3,19 @@ package codecs
 import (
 	"fmt"
 
-	"go-micro.dev/v5/util/container"
-	"go-micro.dev/v5/util/slicemap"
+	"github.com/go-orb/go-orb/util/container"
+	"github.com/go-orb/go-orb/util/slicemap"
 )
 
 // Plugins is the registry for codec plugins.
-var Plugins = container.NewMap[Marshaler]() //nolint:gochecknoglobals
+var Plugins = container.NewPlugins[Marshaler]() //nolint:gochecknoglobals
+
+// Register makes a plugin available by the provided name.
+// If Register is called twice with the same name, it panics.
```

Comment:
Do we really want to panic on double call?

Reply:
That was the behaviour until now, the difference is that panic() is now hidden in container.Plugins().Register.

I've been thinking a lot if that is nice.